### PR TITLE
analytic content for palo alto cdl

### PR DESCRIPTION
**Pull request by spsocprime**
## Proposed Changes
 Added analytic content for Palo Alto CDL data connector

### CODE REVIEW

**Comment by v-sabiraj** [comment]
Hey @spsocprime, We have resolved the issue for PR Validations failure and request you to please update your PR branch with the latest changes from **Azure:master** so that the PR Validations can be executed.... Thanks!!!

**Comment by spsocprime** [comment]
The only issue is with 'Recon' tactic. I guess it is not included in the list (within checking script) cause it is rather new one.
image

**Comment by NikTripathi** [comment]
@aprakash13 Please review analytic rules.

**Comment by v-sabiraj** [comment]
image
image
Please fix the workbooks query, thanks.

**Comment by spsocprime** [comment]
> image image Please fix the workbooks query, thanks.

The query fails because this field name is dynamically created when there are certain types of events from that log source. I guess, the issue exists because there are no such events in your environment

**Comment by v-sabiraj** [comment]
image
The parser has same error, please update.

**Review by v-sabiraj — APPROVED** [review]

**Comment by v-sabiraj** [comment]
Workbooks looks good, can merge once @aprakash13 review and approve, thanks.

**Comment by v-sabiraj** [comment]
Workbooks look good, need to fix validation errors, thanks.

**Comment by v-rucdu** [comment]
> The only issue is with 'Recon' tactic. I guess it is not included in the list (within checking script) cause it is rather new one. image

Hey @spsocprime, we have added the attack tactics, can you please merge the latest from Azure:master and see if Validations are passing now... Thanks!!!

**Comment by v-rucdu** [comment]
@aprakash13 : Can you please take a look at the Analytical Rules... Thanks!!!

File: Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLIncompleteApplicationProtocol.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+id: 7cbd46ce-5121-11ec-bf63-0242ac130002
+name: PaloAlto - Incomplete application protocol
+description: |
+  'Query shows incomplete application protocol'
```
Comment (aprakash13) [inline Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLIncompleteApplicationProtocol.yaml:4]: Can we please check the detection logic and the description. From the logic seems like we are not checking anything related to incomplete application protocol. Please double check.

File: Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLMultiDenyResultbyUser.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+relevantTechniques:
+  - T1190
+  - T1133
+query: |
+  let threshold = 20;
+  PaloAltoCDLEvent
+  | where TimeGenerated > ago(24h)
+  | where DvcAction contains 'deny'
+  | summarize DenyCount = count() by DvcAction, DstUsername
+  | where threshold < DenyCount
```
Comment (aprakash13) [inline Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLMultiDenyResultbyUser.yaml:21]: Wondering if we wanted DenyCount > Threshold here.

File: Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLOutdatedAgentVersions.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+id: 555bf415-e171-4ad2-920f-1a4a96a9644c
+name: PaloAlto - Agent vesions
+description: |
+  'Query shows agent versions'
```
Comment (aprakash13) [inline Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLOutdatedAgentVersions.yaml:4]: Should the description say something about showing agents that are not on the latest version as we are using | where PanOSAgentVersion != cur_ver in the query.

File: Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLOutdatedConfigVersions.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+id: 555bf415-e171-4ad2-920f-1a4a96a9644c
+name: PaloAlto - Outdated config vesions
+description: |
+  'Query shows outdated config vesions'
```
Comment (aprakash13) [inline Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLOutdatedConfigVersions.yaml:4]: Should the query description be the Query shows agent versions. I do not see the below being used in the query for comparison.
let cur_ver = dynamic(['0.1']); //put latest config version here

File: Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLRareApplicationLayerProtocol.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  - InitialAccess
+relevantTechniques:
+  - T1190
+  - T1133
+query: |
+  PaloAltoCDLEvent
+  | where TimeGenerated > ago(24h)
+  | where isnotempty(NetworkApplicationProtocol) 
+  | summarize ApplicationLayerProtocol = count() by NetworkApplicationProtocol
+  | sort by ApplicationLayerProtocol asc
```
Comment (aprakash13) [inline Solutions/PaloAltoCDL/Hunting Queries/PaloAltoCDLRareApplicationLayerProtocol.yaml:20]: How do we determine what is rare. Did we also wanted to do a top 5 or top 10 here.

**Review by aprakash13 — CHANGES_REQUESTED** [review]
Made some suggestions for improvement. Please have a look. Thanks.

**Comment by v-sabiraj** [comment]
@spsocprime , please fix the requested changes, thanks.

**Comment by v-sabiraj** [comment]
@spsocprime ,please update the changes. Thanks.

**Comment by NikTripathi** [comment]
@aprakash13 Please verify the changes made and approve. Thank you.

**Review by aprakash13 — APPROVED** [review]
Modified queries as suggested. Approved

**Pull request merged**